package.json

```diff
@@ -0,0 +1,64 @@
+{
+  "name": "far-nearer",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "directories": {
+    "test": "test"
+  },
+  "scripts": {
+    "dev": "NODE_ENV=development nodemon server/",
+    "start:client": "cd client && npm start",
+    "start": "NODE_ENV=production node server/",
+    "heroku-postbuild": "cd client && npm install && npm run build",
+    "test": "NODE_ENV=test tape test/**/*.js | tap-spec",
+    "lint": "eslint server/",
+    "lint-fix": "eslint server/ --fix",
+    "lint:react": "cd client && npm run lint"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/GSG-G7/far-nearer.git"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/GSG-G7/far-nearer/issues"
+  },
+  "homepage": "https://github.com/GSG-G7/far-nearer#readme",
+  "dependencies": {
+    "airtable": "^0.7.1",
+    "compression": "^1.7.4",
+    "cookie-parser": "^1.4.4",
+    "dotenv": "^8.2.0",
+    "express": "^4.17.1",
+    "express-fileupload": "^1.1.6-alpha.6",
+    "jsonwebtoken": "^8.5.1",
+    "morgan": "^1.9.1",
+    "yup": "^0.27.0"
+  },
+  "devDependencies": {
+    "eslint": "^6.1.0",
+    "eslint-config-airbnb-base": "^14.0.0",
+    "eslint-config-prettier": "^6.4.0",
+    "eslint-config-react-app": "^5.0.2",
+    "eslint-plugin-flowtype": "^4.3.0",
+    "eslint-plugin-import": "^2.18.2",
+    "eslint-plugin-jsx-a11y": "^6.2.3",
+    "eslint-plugin-prettier": "^3.1.1",
+    "eslint-plugin-react": "^7.16.0",
+    "eslint-plugin-react-hooks": "^2.2.0",
```

Comment:
```json
{
    "eslint-config-react-app": "^5.0.2",
    "eslint-plugin-flowtype": "^4.3.0",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-jsx-a11y": "^6.2.3",
    "eslint-plugin-react": "^7.16.0",
    "eslint-plugin-react-hooks": "^2.2.0"
}
```
Are these needed here?

Reply:
Yes, because of eslint config
If we remove them, we need to change vs code settings in each computer

Reply:
but these plugins are for adjusting the code style with react and this `package.json` is for the server :\

Reply:
This what I asked Ahmed Sahatat and Abdel Samad in the workshop and they give me this solution, we have to choose to change vs code settings or add these plugins in the server
because the server start first and have to access these plugins

Reply:
This is the code that we can add it to the settings to enable eslint work without putting the previous plugins
```
 "eslint.workingDirectories": [
    "./client", "./server"
  ]
```

Reply:
It doesn't make sense, then why do we even install them in the `client`? since they are installed in the root and `client` can access them from there.
> because the server start first and have to access these plugins.

Why would the server need to access these plugins?

Reply:
I think that using typescript can solve the problem, but I am not sure

https://www.npmjs.com/package/@typescript-eslint/eslint-plugin

https://www.npmjs.com/package/@typescript-eslint/parser

Reply:
Doesn't make sense either :slightly_smiling_face: 
It's just that I don't even see the problem.

Reply:
@MaiUbeid, @ahmedisam99
If you install them in the root directory then no need to install them in the client
The eslint job is just to make sure the development goes easily, and these packages won't be shipped to the production code. So I think "**in my opinion,**" it doesn't matter if they're in the root directory especially that only the editor plugin will load them not the server nor the react.
The problem appears when you open the root folder in the VS code, the eslint plugin will go look for the eslint plugins "packages" in the node_modules in the root directory.
so you have one of two choices at the moment "maybe later they'll fix this"
either you configure your VS code to recognize the client folder as a root "for eslint" as @MaiUbeid  said, Or you move these packages in the root.
**we never said that you will install them both in the client and the server**

Reply:
@ashatat 
As for loading the plugins and using them, `vscode` has absolutely nothing to do with this process (the extension is the one that does that not `vscode`).
Having this debate of where to put `eslint`'s configuration I just thought of installing `eslint` globally and how would that cause many problems according to what you say.
But it appears that `eslint` can have multiple configuration files and this is the one thing that causes all these problems, so there is actually a solution for that in the documentation of eslint   (**here:** https://eslint.org/docs/user-guide/configuring#using-configuration-files-1)

---

> The problem appears when you open the root folder in the VS code, the eslint plugin will go look for the eslint plugins "packages" in the node_modules in the root directory.

The thing here is that the way you use the term `root directory` isn't clear!! as every directory can be a root directory of its own and `vscode` has nothing to do with that, which actually the one thing that didn't quite hit in my mind in addition for installing unnecessary plugins for the server in its own directory (because when you consider a react-node project you immediately think of two separate projects _- it'll be quite joyful to use a tool like `docker` to completely separate these two -_)

---
@MaiUbeid 
you'll find the solution here: 
- https://eslint.org/docs/user-guide/configuring#using-configuration-files-1
- https://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy